src/server/pfs/cmds/cmds.go

```diff
@@ -1105,74 +1106,50 @@ $ {{alias}} foo@master:path1 bar@master:path2`,
 				if err != nil {
 					return err
 				}
-
-				nI, oI := 0, 0
 				diffCmd := diffCommand(diffCmdArg)
-				for {
-					if nI == len(newFiles) && oI == len(oldFiles) {
-						break
-					}
-					if err := func() (retErr error) {
-						var nFI *pfsclient.FileInfo
-						var oFI *pfsclient.FileInfo
-						switch {
-						case oI == len(oldFiles) || newFiles[nI].File.Path < oldFiles[oI].File.Path:
-							nFI = newFiles[nI]
-							nI++
-						case nI == len(newFiles) || oldFiles[oI].File.Path < newFiles[nI].File.Path:
-							oFI = oldFiles[oI]
-							oI++
-						case newFiles[nI].File.Path == oldFiles[oI].File.Path:
-							nFI = newFiles[nI]
-							nI++
-							oFI = oldFiles[oI]
-							oI++
+				return forEachDiffFile(newFiles, oldFiles, func(nFI, oFI *pfsclient.FileInfo) error {
+					if nameOnly {
+						if newFiles != nil {
```

Comment:
These branches should be checking `nFI` and `oFI` instead of `newFiles` and `oldFile` [sic], I believe.

Reply:
And another good catch!